.github/actions/publish-report/action.yml

```diff
@@ -0,0 +1,131 @@
+---
+name: 'Publish report'
+description: 'Publish report on a PR basis'
+inputs:
+  globFiles:
+    description: 'The glob files to be published'
+    required: true
+  artifact:
+    description: 'The artifact name that contains the files to be published'
+    required: true
+  index:
+    description: 'The index.html to be highlighted in the report'
+    required: true
+  vaultUrl:
+    description: 'Vault URL'
+    required: true
+  vaultRoleId:
+    description: 'Vault role ID'
+    required: true
+  vaultSecretId:
+    description: 'Vault secret ID'
+    required: true
+  vaultSecret:
+    description: 'Vault secret with the format { service-account :"foo" }'
+    required: true
+  bucket:
+    description: 'Google bucket name'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    # as long as it's required to interact with the PR/Issue then this is
+    # the only way when using composite workflows.
+    - id: gh-token
+      uses: elastic/apm-pipeline-library/.github/actions/github-token/@current
+      with:
+        url: ${{ inputs.vaultUrl }}
+        roleId: ${{ inputs.vaultRoleId }}
+        secretId: ${{ inputs.vaultSecretId }}
```

Comment:
not needed the `secrets.GITHUB_TOKEN` can download artifacts from public repos, in case of a private repo you can add a token input to the action.